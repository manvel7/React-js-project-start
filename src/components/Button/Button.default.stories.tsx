import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Button, {IProps} from "./Button"

const meta: Meta<IProps> = {
  title: "Components/Button/Default",
  component: Button,
  argTypes: {
    onClick: {action: "onClick"}
  }
}

const ButtonStory: Story<IProps> = (args) => <Button {...args}>Button</Button>

const Large = ButtonStory.bind({})
Large.args = {
  size: "lg",
  variant: "default"
}

const LargeDisabled = ButtonStory.bind({})
LargeDisabled.args = {
  size: "lg",
  variant: "default",
  isDisabled: true
}

const Medium = ButtonStory.bind({})
Medium.args = {
  size: "md",
  variant: "default"
}

const MediumDisabled = ButtonStory.bind({})
MediumDisabled.args = {
  size: "md",
  variant: "default",
  isDisabled: true
}

const Small = ButtonStory.bind({})
Small.args = {
  size: "sm",
  variant: "default"
}

const SmallDisabled = ButtonStory.bind({})
SmallDisabled.args = {
  size: "sm",
  variant: "default",
  isDisabled: true
}

const Info = ButtonStory.bind({})
Info.args = {
  state: "info"
}

const Success = ButtonStory.bind({})
Success.args = {
  state: "success"
}

const Warn = ButtonStory.bind({})
Warn.args = {
  state: "warn"
}

const Error = ButtonStory.bind({})
Error.args = {
  state: "error"
}

export {
  Large,
  LargeDisabled,
  Medium,
  MediumDisabled,
  Small,
  SmallDisabled,
  Info,
  Success,
  Warn,
  Error
}

export default meta
