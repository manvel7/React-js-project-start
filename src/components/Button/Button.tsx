import {ButtonHTMLAttributes} from "react"
import styled, {css, StyledProps} from "styled-components"
import {h4Mixin, h6Mixin, smallMixin} from "../../styles/typography"

export type ButtonSize = "sm" | "md" | "lg"
export type ButtonVariant = "default" | "outlined" | "text"
export type ButtonState = "info" | "warn" | "success" | "error"

export interface IProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  isDisabled?: boolean
  size?: ButtonSize
  variant?: ButtonVariant
  state?: ButtonState
}

const lgMixin = css<StyledProps<IProps>>`
  border-radius: 0.8rem;
  padding: 0.8rem 1.6rem;
  ${h4Mixin<IProps>()}
`

const smMixin = css<StyledProps<IProps>>`
  border-radius: 0.3rem;
  padding: 0.2rem 1.6rem;
  ${smallMixin<IProps>()}
`

const mdMixin = css<StyledProps<IProps>>`
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  ${h6Mixin<IProps>()}
`

const outlinedMixin = css<StyledProps<IProps>>`
  background-color: transparent;
  border: 0.1rem solid
    ${({isDisabled, state, theme}) => {
      if (isDisabled) {
        return theme.palette.disabled
      }

      switch (state) {
        case "error":
          return theme.palette.alert.error
        case "warn":
          return theme.palette.alert.warn
        case "success":
          return theme.palette.alert.success
        case "info":
        default:
          return theme.palette.primary
      }
    }};
  box-sizing: border-box;
  color: ${({isDisabled, theme}) =>
    isDisabled ? theme.palette.text.darkDimmed : theme.palette.primary};
`

const textMixin = css<StyledProps<IProps>>`
  background-color: transparent;
  color: ${({isDisabled, theme}) =>
    isDisabled ? theme.palette.text.darkDimmed : theme.palette.primary};
`

const defaultMixin = css<StyledProps<IProps>>`
  background-color: ${({isDisabled, state, theme}) => {
    if (isDisabled) {
      return theme.palette.background.disabled
    }

    switch (state) {
      case "error":
        return theme.palette.alert.error
      case "warn":
        return theme.palette.alert.warn
      case "success":
        return theme.palette.alert.success
      case "info":
      default:
        return theme.palette.primary
    }
  }};
  color: ${({isDisabled, theme}) =>
    isDisabled ? theme.palette.text.darkDimmed : theme.palette.text.white};
`

const infoMixin = css<StyledProps<IProps>>`
  color: ${({theme}) => theme.palette.alert.info};
`

const warnMixin = css<StyledProps<IProps>>`
  color: ${({theme}) => theme.palette.alert.warn};
`

const errorMixin = css<StyledProps<IProps>>`
  color: ${({theme}) => theme.palette.alert.error};
`

const successMixin = css<StyledProps<IProps>>`
  color: ${({theme}) => theme.palette.alert.success};
`

const Button = styled.button<IProps>`
  border: 0.2rem solid transparent;
  cursor: pointer;

  ${({size}) => {
    switch (size) {
      case "lg":
        return lgMixin
      case "sm":
        return smMixin
      case "md":
      default:
        return mdMixin
    }
  }}

  ${({variant}) => {
    switch (variant) {
      case "outlined":
        return outlinedMixin
      case "text":
        return textMixin
      case "default":
      default:
        return defaultMixin
    }
  }}

  ${({state, variant}) => {
    if (!variant || variant === "default") {
      return ""
    }

    switch (state) {
      case "info":
        return infoMixin
      case "success":
        return successMixin
      case "warn":
        return warnMixin
      case "error":
        return errorMixin
      default:
        return ""
    }
  }}
`

export default Button
