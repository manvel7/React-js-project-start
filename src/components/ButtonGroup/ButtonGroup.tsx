import {HTMLAttributes} from "react"
import styled from "styled-components"

interface IProps extends HTMLAttributes<HTMLDivElement> {}

const ButtonGroup = styled.div<IProps>`
  display: inline-flex;
  flex-wrap: no-wrap;

  button {
    &:first-child {
      border-bottom-right-radius: 0;
      border-top-right-radius: 0;
    }

    &:last-child {
      border-bottom-left-radius: 0;
      border-top-left-radius: 0;
    }

    &:not(:first-child):not(:last-child) {
      border-radius: 0;
    }

    &.toggle__button--inactive {
      border-color: ${({theme}) => theme.palette.border};
    }

    &.toggle__button--active {
      font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
    }
  }
`

export default ButtonGroup
