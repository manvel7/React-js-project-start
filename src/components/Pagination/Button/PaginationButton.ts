import {ButtonHTMLAttributes} from "react"
import styled, {css, StyledProps} from "styled-components"
import {h4Mixin, h5Mixin} from "../../../styles/typography"

export type ButtonSize = "md" | "lg"
export type ButtonVariant = "normal" | "circle"

interface IProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  isActive?: boolean
  size?: ButtonSize
  variant?: ButtonVariant
}

const lgMixin = css<StyledProps<IProps>>`
  ${h4Mixin<IProps>()}
`

const mdMixin = css<StyledProps<IProps>>`
  ${h5Mixin<IProps>()}
`

const normalMixin = css<StyledProps<IProps>>`
  border-radius: 0.4rem;
`

const circleMixin = css<StyledProps<IProps>>`
  /*width is not fixed, so 50% doesn't work*/
  border-radius: 10rem;
`

const activeMixin = css<StyledProps<IProps>>`
  background-color: ${({theme}) => theme.palette.primary};
  color: ${({theme}) => theme.palette.text.white};
`

const PaginationButton = styled.button<IProps>`
  background-color: ${({theme}) => theme.palette.background.tinted};
  border: none;
  color: ${({theme}) => theme.palette.primary};
  cursor: pointer;
  margin: 0 0.4rem;
  min-height: 3.2rem;
  min-width: 3.2rem;
  padding: 0.4rem 0.8rem;

  &[disabled] {
    cursor: initial;
  }

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    margin-right: 0;
  }

  ${({size}) => {
    switch (size) {
      case "lg":
        return lgMixin
      case "md":
      default:
        return mdMixin
    }
  }}

  ${({variant}) => {
    switch (variant) {
      case "circle":
        return circleMixin
      case "normal":
      default:
        return normalMixin
    }
  }}

  ${({isActive}) => (isActive ? activeMixin : "")}
`

PaginationButton.defaultProps = {
  variant: "normal",
  isActive: false
}

export default PaginationButton
