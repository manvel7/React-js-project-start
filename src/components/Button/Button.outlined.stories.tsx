import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Button, {IProps} from "./Button"

const meta: Meta<IProps> = {
  title: "Components/Button/Outlined",
  component: Button,
  argTypes: {
    onClick: {action: "onClick"}
  }
}

const ButtonStory: Story<IProps> = (args) => <Button {...args}>Button</Button>

const Large = ButtonStory.bind({})
Large.args = {
  size: "lg",
  variant: "outlined"
}

const LargeDisabled = ButtonStory.bind({})
LargeDisabled.args = {
  size: "lg",
  variant: "outlined",
  isDisabled: true
}

const Medium = ButtonStory.bind({})
Medium.args = {
  size: "md",
  variant: "outlined"
}

const MediumDisabled = ButtonStory.bind({})
MediumDisabled.args = {
  size: "md",
  variant: "outlined",
  isDisabled: true
}

const Small = ButtonStory.bind({})
Small.args = {
  size: "sm",
  variant: "outlined"
}

const SmallDisabled = ButtonStory.bind({})
SmallDisabled.args = {
  size: "sm",
  variant: "outlined",
  isDisabled: true
}

const Info = ButtonStory.bind({})
Info.args = {
  variant: "outlined",
  state: "info"
}

const Success = ButtonStory.bind({})
Success.args = {
  variant: "outlined",
  state: "success"
}

const Warn = ButtonStory.bind({})
Warn.args = {
  variant: "outlined",
  state: "warn"
}

const Error = ButtonStory.bind({})
Error.args = {
  variant: "outlined",
  state: "error"
}

export {
  Large,
  LargeDisabled,
  Medium,
  MediumDisabled,
  Small,
  SmallDisabled,
  Info,
  Success,
  Warn,
  Error
}

export default meta
