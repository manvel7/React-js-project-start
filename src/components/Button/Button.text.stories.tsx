import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Button, {IProps} from "./Button"

const meta: Meta<IProps> = {
  title: "Components/Button/Text",
  component: Button,
  argTypes: {
    onClick: {action: "onClick"}
  }
}

const ButtonStory: Story<IProps> = (args) => <Button {...args}>Button</Button>

const Large = ButtonStory.bind({})
Large.args = {
  size: "lg",
  variant: "text"
}

const LargeDisabled = ButtonStory.bind({})
LargeDisabled.args = {
  size: "lg",
  variant: "text",
  isDisabled: true
}

const Medium = ButtonStory.bind({})
Medium.args = {
  size: "md",
  variant: "text"
}

const MediumDisabled = ButtonStory.bind({})
MediumDisabled.args = {
  size: "md",
  variant: "text",
  isDisabled: true
}

const Small = ButtonStory.bind({})
Small.args = {
  size: "sm",
  variant: "text"
}

const SmallDisabled = ButtonStory.bind({})
SmallDisabled.args = {
  size: "sm",
  variant: "text",
  isDisabled: true
}

const Info = ButtonStory.bind({})
Info.args = {
  variant: "text",
  state: "info"
}

const Success = ButtonStory.bind({})
Success.args = {
  variant: "text",
  state: "success"
}

const Warn = ButtonStory.bind({})
Warn.args = {
  variant: "text",
  state: "warn"
}

const Error = ButtonStory.bind({})
Error.args = {
  variant: "text",
  state: "error"
}

export {
  Large,
  LargeDisabled,
  Medium,
  MediumDisabled,
  Small,
  SmallDisabled,
  Info,
  Success,
  Warn,
  Error
}

export default meta
