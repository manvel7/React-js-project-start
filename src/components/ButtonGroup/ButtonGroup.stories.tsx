import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Button from "../Button"
import ButtonGroup from "../ButtonGroup/ButtonGroup"

const meta: Meta = {
  title: "Components/ButtonGroup",
  component: ButtonGroup,
  argTypes: {}
}

const ButtonGroupStory: Story = (args) => (
  <ButtonGroup {...args}>
    <Button size="sm">First</Button>
    <Button size="sm">Second</Button>
    <Button size="sm">Third</Button>
  </ButtonGroup>
)

const Regular = ButtonGroupStory.bind({})

export {Regular}

export default meta
