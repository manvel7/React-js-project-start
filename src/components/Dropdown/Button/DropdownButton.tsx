import React, {FunctionComponent} from "react"
import {faCaretDown} from "@fortawesome/pro-solid-svg-icons"
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome"
import Button from "../../Button"
import ButtonGroup from "../../ButtonGroup"
import {IProps as IDropdownProps} from "../Dropdown"

interface IProps extends IDropdownProps {
  isOpened: boolean
  onToggleOpen: () => void
}

const DropdownButton: FunctionComponent<IProps> = ({
  size,
  onButtonClick,
  onToggleOpen,
  title,
  isOpened
}) => (
  <ButtonGroup>
    <Button size={size} onClick={onButtonClick}>
      {title}
    </Button>
    <Button size={size} onClick={onToggleOpen}>
      <FontAwesomeIcon
        icon={faCaretDown}
        size="sm"
        rotation={isOpened ? 180 : undefined}
      />
    </Button>
  </ButtonGroup>
)

export default DropdownButton
