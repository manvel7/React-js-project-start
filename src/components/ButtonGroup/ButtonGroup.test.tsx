import React from "react"
import {render} from "@testing-library/react"
import {UiProvider} from "../../styles"
import Button from "../Button"
import ButtonGroup from "./ButtonGroup"

describe("ButtonGroup", () => {
  describe("Regular", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <ButtonGroup>
            <Button>First</Button>
            <Button>Second</Button>
            <Button>Third</Button>
          </ButtonGroup>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })
})
