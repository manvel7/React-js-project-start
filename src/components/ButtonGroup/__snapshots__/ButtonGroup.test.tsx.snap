// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup Regular should match snapshot 1`] = `
.c1 {
  border: 0.2rem solid transparent;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.6rem 1.6rem;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
  background-color: #001254;
  color: #FEFEFE;
}

.c1 b,
.c1 strong {
  font-weight: 800;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
}

.c0 button:first-child {
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
}

.c0 button:last-child {
  border-bottom-left-radius: 0;
  border-top-left-radius: 0;
}

.c0 button:not(:first-child):not(:last-child) {
  border-radius: 0;
}

.c0 button.toggle__button--inactive {
  border-color: #E6E6E6;
}

.c0 button.toggle__button--active {
  font-weight: 800;
}

<div
  class="c0"
>
  <button
    class="c1"
  >
    First
  </button>
  <button
    class="c1"
  >
    Second
  </button>
  <button
    class="c1"
  >
    Third
  </button>
</div>
`;
